pkg/resource/certificate/hooks.go

```diff
@@ -58,7 +64,70 @@ func validatePublicValidationOptions(
 	return nil
 }
 
+// maybeImportCertificate imports a certificate into ACM if Spec.Certificate is set.
+func (rm *resourceManager) maybeImportCertificate(ctx context.Context, r *resource) (*resource, bool, error) {
+	certSpec := r.ko.Spec
+	if certSpec.Certificate != nil {
+		if certSpec.DomainName != nil || len(certSpec.DomainValidationOptions) > 0 || certSpec.KeyAlgorithm != nil ||
+			len(certSpec.SubjectAlternativeNames) > 0 || certSpec.Options != nil {
+			return nil, false, ackerr.NewTerminalError(errors.New("cannot set fields used for requesting a certificate when importing a certificate"))
+		}
+		input, err := rm.newImportCertificateInput(ctx, r)
+		if err != nil {
+			return nil, false, err
+		}
+		if len(input.PrivateKey) == 0 {
+			return nil, false, ackerr.NewTerminalError(errors.New("privateKey is required when importing a certificate"))
+		}
+		created, err := rm.importCertificate(ctx, r, input)
+		if err != nil {
+			return nil, false, err
+		}
+		return created, true, nil
```

Comment:
Do we need the boolean in the outputs? looks like we return true only when `created` is returned.

Reply:
While the caller can also use `nil` for checking if this is for importing an existing certificate, a separate boolean return value makes the intent clearer and the calling code more readable.